app/javascript/components/assessment/steps/SymptomsAssessment.js

```diff
@@ -19,15 +19,36 @@ class SymptomsAssessment extends React.Component {
     };
   }
 
-  handleChange = event => {
-    let value = event.target.type === 'checkbox' ? event.target.checked : event.target.value;
+  handleChange = (event, value) => {
     let report = this.state.reportState;
     let field_id = event.target.id.split('_idpre')[0];
     Object.values(report.symptoms).find(symp => symp.name === field_id).value = value;
     this.setState({ reportState: report });
     this.updateBoolSymptomCount();
   };
 
+  handleBoolChange = event => {
+    let value = event.target.checked;
+    this.handleChange(event, value);
+  };
+
+  handleIntChange = event => {
+    const validInputs = ['', '-'];
+    if (
+      validInputs.includes(event?.target?.value) ||
```

Comment:
Should put `event?.target?.value` in a variable since you use it a bunch here. I also think if it's `undefined` or `null` for whatever reason you should avoid checking `includes` on it. I'd ask you add a comment here explaining these checks as well.

Reply:
I updated the structure of this conditional statement a bit and included a comment explaining why each of the checks are needed. And I split out `event?.target?.value` into a variable. I did a similar thing for the `handleFloatChange()` method. Let me know what you think.